### DISCUSSION TITLE
Addition of Llama2

### DISCUSSION THREAD

**Discussion by pavankalyan066** [sinaptik-ai/pandas-ai#528] [Ideas]
I like the features of this library and curious to know about the update regarding addition of LLama2 model.

I have tried with langchain features such as create_csv_agent etc, but the processing results takes time.

Thank you.

**Comment by gventuri** [comment]
Hi @pavankalyan066 thanks a lot for the kind words. At the moment, we support LLaMa2 via https://docs.pandas-ai.com/en/latest/LLMs/llms/#huggingface-via-text-generation.

The only downside is at the moment there are not free APIs doing that, so one could either try with a local version or deploying LLaMa2 to a private cloud (as an alternative of using Hugging Face inference API!
